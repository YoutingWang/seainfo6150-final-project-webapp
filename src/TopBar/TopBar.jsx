import React from 'react';
import { useHistory } from "react-router-dom";
import styles from "./TopBar.module.css";

const TopBar = (props) => {
    const history = useHistory();
    function handleClickHome() {
        history.push("/");
    }
    function handleClickAbout() {
        history.push("/about");
    }
    function handleClickContact() {
        history.push("/contact");
    }
    return (
        <div className={styles.topBar}>
            <nav>
                <button type="button" onClick={handleClickHome}>
                    Home
                </button>
                <button type="button" onClick={handleClickAbout}>
                    About
                </button>
                <button type="button" onClick={handleClickContact}>
                    Contact us
                </button>
            </nav>
        </div>
        
    )
}

export default TopBar
