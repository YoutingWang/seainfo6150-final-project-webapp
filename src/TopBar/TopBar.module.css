.topBar {
    display: flex;
    align-items: center;
    justify-content: center;
}