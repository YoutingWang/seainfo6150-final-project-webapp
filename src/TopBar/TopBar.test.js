import React from "react";
import {
    render
} from "@testing-library/react";
import TopBar from "./TopBar.jsx";

it("renders the home component correctly", () => {
    const {
        container
    } = render( < TopBar / > );
    expect(container).toMatchSnapshot();
});