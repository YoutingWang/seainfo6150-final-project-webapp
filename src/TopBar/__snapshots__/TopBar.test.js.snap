// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home component correctly 1`] = `
<div>
  <div
    class="topBar"
  >
    <nav>
      <button
        type="button"
      >
        Home
      </button>
      <button
        type="button"
      >
        About
      </button>
      <button
        type="button"
      >
        Contact us
      </button>
    </nav>
  </div>
</div>
`;
